src/expr/src/expr/expr_binary_nonnull.rs

```diff
@@ -206,6 +207,40 @@ macro_rules! gen_binary_expr_atm {
     };
 }
 
+/// `gen_binary_expr_shift` is similar to `gen_binary_expr_atm`.
+///  `shift` means arithmetic shift here.
+/// They are differentiate cuz shift operation does not support for float datatype.
```

Comment:
* `shift` -> `bitwise`
* `<<` and `>>` have different typing rules from `&`/`|`/`#` in [PostgreSQL](https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_operator.dat).
  * The right side can only be `int16` or `int32` but not `int64`.
  * The return type is always the same as the left side, rather than the bigger of the two.
```
test=# select pg_typeof(1::smallint << 2::integer);
 pg_typeof 
-----------
 smallint
(1 row)

test=# select pg_typeof(1::smallint << 2::bigint);
ERROR:  operator does not exist: smallint << bigint
LINE 1: select pg_typeof(1::smallint << 2::bigint);
                                     ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.

```

Reply:
Thank you for the comment. So to implement this, does this means that I only have to accept T1 and T2 only. So I need to defined a macro rules that pass through T1 and T2 only.

Reply:
Yes, only T1 and T2.